### PR TITLE
add ci testing and update publishing instructions

### PR DESCRIPTION
**Pull request by tatthurs**

### CODE REVIEW

File: lib/fast_jsonapi/object_serializer.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@
 require 'active_support/concern'
 require 'active_support/inflector'
 require 'active_support/core_ext/numeric/time'
+require "active_support/core_ext/object"
```
Comment (tatthurs) [inline lib/fast_jsonapi/object_serializer.rb:8]:
```suggestion
require 'active_support/core_ext/object'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,53 +1,29 @@
...
+      - uses: actions/checkout@v2
 
-    - name: Runs code QA and tests
-      run: bundle exec rake
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
 
-    - name: Publish to Rubygems
```
Comment (lexahu) [inline .github/workflows/ci.yml:42]: So publishing this gem was a part of CI? or merging to master? and now we are moving it to be done manually locally right?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,53 +1,29 @@
...
-
-    - name: Install bundler 2.4+ for modern Rubies
-      if: ${{ matrix.ruby != 2.4 }}
-      run: |
-        gem install -q bundler
-
-    - name: Run bundle install
-      run: |
-        bundle install
+      - uses: actions/checkout@v2
```
Comment (lexahu) [inline .github/workflows/ci.yml]: Why is this downgraded from actions/checkout@v3 to actions/checkout@v2?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,53 +1,29 @@
...
 jobs:
-  tests:
+  test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        ruby: [2.4, 2.7, '3.0', 3.1, 3.2, truffleruby-head]
-
+        ruby: ["2.7", "3.0", "3.1", "3.2", "3.3"]
```
Comment (lexahu) [inline .github/workflows/ci.yml:16]: Why are we removing truffleruby-head?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -1,53 +1,29 @@
...
-        chmod 0600 $HOME/.gem/credentials
-        printf -- "---\n:rubygems_api_key: Bearer ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
-        gem build *.gemspec
-        gem push *.gem
-      env:
-        GEM_HOST_API_KEY: ${{secrets.RUBYGEMS_AUTH_TOKEN}}
+      - run: |
+          bundle exec rake spec
+        env:
+          RAILS_ENV: test
```
Comment (lexahu) [inline .github/workflows/ci.yml]: is this needed if there is no rails?

File: lib/fast_jsonapi/object_serializer.rb (hunk 1 above)
Comment (lexahu) [reply lib/fast_jsonapi/object_serializer.rb:8]: why does this have double quotes?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (tatthurs) [reply .github/workflows/ci.yml:16]: we don't use truffle ruby

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tatthurs) [reply .github/workflows/ci.yml:42]: the upstream gem had this setup, but I didn't wire this up for our gem forks. we could consider that follow upfor all our gems in github down the road.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (tatthurs) [reply .github/workflows/ci.yml]: good catch -- copy paste, removed.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (tatthurs) [reply .github/workflows/ci.yml]: copy paste -- updated to latest (v4)

File: Rakefile [hunk 6]
```diff
@@ -2,6 +2,14 @@ require 'bundler/gem_tasks'
 require 'rspec/core/rake_task'
 require 'rubocop/rake_task'
 
+desc 'Build the package and publish it to rubygems.pkg.github.com'
+task publish: :build do
+  # Requires local setup of personal access token, see:
+  # 1. https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-rubygems-registry#authenticating-with-a-personal-access-token
+  system("gem push --key github --host https://rubygems.pkg.github.com/art19 " \
```
Comment (rortian) [inline Rakefile:9]: stupid nitty: I like multiline quote styles over this

**Review by rortian — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -1,53 +1,27 @@
...
 jobs:
-  tests:
+  test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        ruby: [2.4, 2.7, '3.0', 3.1, 3.2, truffleruby-head]
-
+        ruby: ["2.7", "3.0", "3.1", "3.2", "3.3"]
```
Comment (lexahu) [inline .github/workflows/ci.yml:16]: nit: remove 2.7 now?

**Review by lexahu — APPROVED** [review]

**Pull request merged**